dask/dataframe/utils.py

```diff
@@ -497,18 +497,18 @@ def _nonempty_series(s, idx=None):
 
 def is_dataframe_like(df):
     """ Looks like a Pandas DataFrame """
-    return set(dir(df)) > {'dtypes', 'columns', 'groupby', 'head'}
+    return set(dir(df)) > {'dtypes', 'columns', 'groupby', 'head'} and not isinstance(df, type)
```

Comment:
Anyone have a cleaner suggestion for this?

Reply:
How about
```
import inspect 

def _is_like(obj, includes, excludes=()):
    return (not inspect.isclass(obj)
            and all(hasattr(obj, attr) for attr in includes)
            and not any(hasattr(obj, attr) for attr in excludes))


def is_dataframe_like(df):
    """ Looks like a Pandas DataFrame """
    return _is_like(df, ('dtypes', 'columns', 'groupby', 'head'))


def is_series_like(s):
    """ Looks like a Pandas Series """
    return _is_like(s, ('name', 'dtype', 'groupby', 'head'))


def is_index_like(s):
    """ Looks like a Pandas Index """
    return _is_like(s, ('name', 'dtype'), ('head',))
```

Reply:
You could check for `not hasattr(df, 'mro')` to exclude types, since all classes have an `mro` method.

Reply:
I think I'd like to pass on the more generic version.  There are only three things here and I think it's probably simpler to keep things flat.

The `hasattr(df, 'mro')` check seems equivalent to the isinstance call.

Mostly I was concerned about the case where an instantiated dataframe was also a type, but I think I'll just pass on this case.